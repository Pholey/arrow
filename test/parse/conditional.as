// Statement (Expression)
if condition {
  // [...]
} else if other_condition {
  // [...]
} else {
  // [...]
}

if condition {
  // [...]
}

// Expression (will look a lot nicer once `;` is optional)
let name: int32 = if a > b {
  30;
} else {
  0xdeadbeef;
};
