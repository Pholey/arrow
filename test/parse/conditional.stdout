{
  "tag": "Module",
  "span": "test/parse/conditional.as:1-19",
  "statements": [
    {
      "tag": "Conditional",
      "span": "test/parse/conditional.as:2-8",
      "branches": [
        {
          "tag": "Branch",
          "span": "test/parse/conditional.as:2-4",
          "condition": {
            "tag": "Identifier",
            "span": "test/parse/conditional.as:2:4-13",
            "text": "condition"
          },
          "block": {
            "tag": "Block",
            "span": "test/parse/conditional.as:2-4",
            "statements": []
          }
        },
        {
          "tag": "Branch",
          "span": "test/parse/conditional.as:4-6",
          "condition": {
            "tag": "Identifier",
            "span": "test/parse/conditional.as:4:11-26",
            "text": "other_condition"
          },
          "block": {
            "tag": "Block",
            "span": "test/parse/conditional.as:4-6",
            "statements": []
          }
        }
      ],
      "otherwise": {
        "tag": "Block",
        "span": "test/parse/conditional.as:6-8",
        "statements": []
      }
    },
    {
      "tag": "Conditional",
      "span": "test/parse/conditional.as:10-12",
      "branches": [
        {
          "tag": "Branch",
          "span": "test/parse/conditional.as:10-12",
          "condition": {
            "tag": "Identifier",
            "span": "test/parse/conditional.as:10:4-13",
            "text": "condition"
          },
          "block": {
            "tag": "Block",
            "span": "test/parse/conditional.as:10-12",
            "statements": []
          }
        }
      ],
      "otherwise": null
    },
    {
      "tag": "Variable",
      "span": "test/parse/conditional.as:15-19",
      "name": "name",
      "mutable": false,
      "type": {
        "tag": "TypeName",
        "span": "test/parse/conditional.as:15:11-16",
        "name": {
          "tag": "Identifier",
          "span": "test/parse/conditional.as:15:11-16",
          "text": "int32"
        }
      },
      "initializer": {
        "tag": "Conditional",
        "span": "test/parse/conditional.as:15-19",
        "branches": [
          {
            "tag": "Branch",
            "span": "test/parse/conditional.as:15-17",
            "condition": {
              "tag": "GreaterThan",
              "span": "test/parse/conditional.as:15:24-27",
              "lhs": {
                "tag": "Identifier",
                "span": "test/parse/conditional.as:15:22-23",
                "text": "a"
              },
              "rhs": {
                "tag": "Identifier",
                "span": "test/parse/conditional.as:15:26-27",
                "text": "b"
              }
            },
            "block": {
              "tag": "Block",
              "span": "test/parse/conditional.as:15-17",
              "statements": [
                {
                  "tag": "Integer",
                  "span": "test/parse/conditional.as:16:3-5",
                  "value": "30"
                }
              ]
            }
          }
        ],
        "otherwise": {
          "tag": "Block",
          "span": "test/parse/conditional.as:17-19",
          "statements": [
            {
              "tag": "Integer",
              "span": "test/parse/conditional.as:18:3-13",
              "value": "3735928559"
            }
          ]
        }
      }
    }
  ]
}
